compiler/noirc_frontend/src/monomorphization/ast.rs

```diff
@@ -530,6 +531,76 @@ impl Type {
             _ => None,
         }
     }
+
+    /// Convert the type back into a HIR equivalent (not necessarily the original HIR type).
+    ///
+    /// Aims to maintain parity with [Monomorphizer::convert_type](crate::monomorphization::Monomorphizer::convert_type).
+    pub fn to_hir_type(&self) -> hir_def::types::Type {
```

Comment:
I am a bit worried that this is more to maintain over time. The Hir -> Mast transformation wasn't meant to be reversible in general.

Reply:
It had to be done in order for AST generation, in order to be able to produce a function signature needed for the artifact. Perhaps we can keep it private, and only expose it with a feature flag for the AST fuzzer to use.